### PR TITLE
Bug Fix: Strikethrough markdown within a link is not rendered in live comments

### LINKED ISSUE #13935: Strikethrough markdown within a link is not rendered in live comments (but works fine elsewhere)
**Describe the bug**
While using [Strikethrough markdown](https://github.com/adam-p/markdown-here/wiki/Markdown-Here-Cheatsheet#emphasis) within [link text](https://github.com/adam-p/markdown-here/wiki/Markdown-Here-Cheatsheet#links), I discovered the strikethrough is not rendered **in live comment views only**; this appears to work elsewhere (comment previews, posts, and post previews).

**To Reproduce**

Add strikethrough text within a link to a comment:
```
[~~strikethrough~~](https://www.example.com)
```

This looks fine in posts, post previews, and comment previews:
image
image
image

...but not in a submitted comment:
image

**Expected behavior**

Strikethrough markdown within link text is rendered correctly.

**Desktop (please complete the following information):**

- OS, version: Fedora Core 33
- Browser, version: Firefox 88.0.1 and Chromium 91.0.4472.77

### PR DESCRIPTION
**Pull request by squarebat**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [x] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description
@andygeorge reported that strikethrough line was not rendered within links in live comments, however it worked elsewhere in posts, post previews and comment previews. The issue was that the html tag for strikethrough ```<del>``` was being stripped out by ```shorten_urls``` method. This PR modifies the ```shorten_urls``` method to not strip out the del tag. Further changes might be required to the method as and when unexpected bugs occur.

## Related Tickets & Documents
Resolves #13935
Note: PR unintentionally modifies ```schema.rb``` due to migrations. 

## QA Instructions, Screenshots, Recordings

markdown for strikethrough within link.
```
[~~strikethrough~~](https://www.example.com)
```
Correctly rendered in live comments after bug fix
strikethrough
## Added/updated tests?

- [ ] Yes
- [x] No, and this is why: a minor fix made in ```shorten_urls``` method which does not require testing
- [ ] I need help with writing tests

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

- [ ] I've updated the [Developer Docs](https://docs.forem.com) and/or
      [Admin Guide](https://admin.forem.com/), or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
- [ ] I've updated the README or added inline documentation
- [ ] I will share this change in a [Changelog](https://forem.dev/t/changelog)
      or in a [forem.dev](http://forem.dev) post
- [ ] I will share this change internally with the appropriate teams
- [ ] I'm not sure how best to communicate this change and need help
- [ ] This change does not need to be communicated, and this is why not: _please
      replace this line with details on why this change doesn't need to be
      shared_

## [optional] Are there any post deployment tasks we need to perform?
None

### CODE REVIEW

File: db/schema.rb [hunk 1]
```diff
@@ -1358,7 +1358,7 @@
     t.datetime "last_article_at", default: "2017-01-01 05:00:00"
     t.datetime "last_comment_at", default: "2017-01-01 05:00:00"
     t.datetime "last_followed_at"
-    t.datetime "last_moderation_notification", default: "2017-01-01 05:00:00"
+    t.datetime "last_moderation_notification", default: "2017-01-01 00:00:00"
```
Comment (djuber) [inline db/schema.rb:1361]: I think these schema.rb changes can be safely removed from the commit, it's probably something from your initial setup caused these changes (they're unrelated to your change above and not needed to render strikethrough)

File: db/schema.rb [hunk 2]
```diff
@@ -1439,9 +1439,9 @@
 
   create_table "users_gdpr_delete_requests", force: :cascade do |t|
     t.datetime "created_at", precision: 6, null: false
-    t.string "email"
+    t.string "email", null: false
     t.datetime "updated_at", precision: 6, null: false
-    t.integer "user_id"
+    t.integer "user_id", null: false
```
Comment (jgaskins) [inline db/schema.rb:1444]: Can you exclude the changes to this file from the PR? It looks like they got committed accidentally.

We've been getting a lot of PRs with these lines changed in them, so I checked and DEV's production DB _does_ have these constraints, so it was probably a mistake that these didn't get committed before, but we'll fix it separately.

**Review by jgaskins — CHANGES_REQUESTED** [review]
Thank you for this fix! Looks good from my perspective after a quick touch-up on the diff to remove the DB schema changes.

**Review by djuber — APPROVED** [review]
Thank you, @squarebat! I tested locally - both articles and comments render strikethrough links correctly. Approved, assuming changes requested (removing the schema change) are done.

File: db/schema.rb (hunk 2 above)
Comment (squarebat) [reply db/schema.rb:1444]: Yeah, I'll do that. The changes were unintentional but I wasn't sure if I should revert them so pushed it anyway.

**Comment by squarebat** [comment]
The required changes are done. I restored ```schema.rb``` to its original state.

**Review by jgaskins — APPROVED** [review]
Sweeeeeet! Thanks @squarebat!!

File: db/schema.rb (hunk 2 above)
Comment (jgaskins) [reply db/schema.rb]: All good. It's probably our fault. :joy:

**Review by andygeorge — APPROVED** [review]

**Pull request merged**